src/ControlSystem/FunctionOfTime.hpp

```diff
@@ -0,0 +1,63 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Utilities/EqualWithinRoundoff.hpp"
+#include "Utilities/MakeArray.hpp"
+
+/// \ingroup ControlSystemGroup
+/// Contains functions of time.
+namespace FunctionOfTime {
+/// \ingroup ControlSystemGroup
+/// A function that has a piecewise-constant `MaxDeriv`th derivative.
+template <size_t MaxDeriv>
+class PiecewisePolynomial {
+ public:
+  // There exists a DataVector for each deriv order that contains
+  // the values of that deriv order for all components.
+  using value_type = std::array<DataVector, MaxDeriv + 1>;
```

Comment:
Does the type alias actually need to be public? STL containers have a value_type that's public, but this class doesn't fall under that concept.